tests/training/multi_gpu_test.py

```diff
@@ -0,0 +1,68 @@
+# pylint: disable=no-self-use,invalid-name
+
+
+from copy import deepcopy
+import keras.backend as K
+import tensorflow
+from deep_qa.training.multi_gpu import pin_variable_device_scope
+from deep_qa.common.params import Params
+from deep_qa.models.text_classification import ClassificationModel
+from ..common.test_case import DeepQaTestCase
+
+
+class TestMultiGpu(DeepQaTestCase):
+    # pylint: disable=protected-access
+
+    def setUp(self):
+        super(TestMultiGpu, self).setUp()
+        self.write_true_false_model_files()
+        self.args = Params({
+                'embedding_dim': {'words': 4, 'characters': 2},
+                'batch_size': 8,
+                'num_gpus': 2,
+                'save_models': True,
+                'show_summary_with_masking_info': True,
+        })
+
+    def test_model_can_train_and_load(self):
+        self.ensure_model_trains_and_loads(ClassificationModel, self.args)
```

Comment:
If you set `soft_device_placement=True` in the tensorflow session, tf won't complain if it can't allocate things exactly on the device you specify, so in this case, we'd be building 2 models on the cpu. If we remove this option, it would crash. I think generally it's recommended to allow tensorflow this flexibility, as it does some optimisiation in terms of device placements, like putting embedding lookups on the same device as the embedding etc.